### PR TITLE
dev.hdos.HDOS

### PR DESCRIPTION
**Pull request by Brainfree**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Review by TheRealMatthew4 — APPROVED** [review]

**Comment by hfiguiere** [comment]
Submission PR should be against the `new-pr` branch.

File: dev.hdos.HDOS.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+    <id>dev.hdos.HDOS.desktop</id>
```
Comment (hfiguiere) [inline dev.hdos.HDOS.appdata.xml:3]:
```suggestion
    <id>dev.hdos.HDOS</id>
```

File: dev.hdos.HDOS.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--device=dri",
+        "--socket=pulseaudio",
+        "--env=JAVA_HOME=/app/jre",
+        "--talk-name=org.freedesktop.Notifications",
+        "--filesystem=xdg-run/app/com.discordapp.Discord:create",
+        "--filesystem=~/hdos:create",
```
Comment (hfiguiere) [inline dev.hdos.HDOS.json:20]: this is redundant with (broad) `home` permission

File: dev.hdos.HDOS.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://gitlab.gnome.org/GNOME/libnotify/-/archive/0.8.1/libnotify-0.8.1.tar.gz",
+                    "sha256": "7c0b252edecbf08db50d775f9e720ecc03c742fb97c25f3966a8b7a4bedf8133"
+                }
+            ],
+            "cleanup": [
+                "/include",
+                "/lib/pkgconfig",
+                "/lib/debug",
```
Comment (hfiguiere) [inline dev.hdos.HDOS.json:47]:
```suggestion
```

File: dev.hdos.HDOS.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                "/share/gir-1.0"
+            ]
+        },
+        {
+            "name": "hdos",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -D hdos /app/bin/hdos",
+                "install -D hdos-launcher.jar /app/share/hdos-launcher.jar",
+                "install -Dm644 dev.hdos.HDOS.png /app/share/icons/hicolor/128x128/apps/dev.hdos.HDOS.png",
```
Comment (hfiguiere) [inline dev.hdos.HDOS.json:59]:
```suggestion
                "install -Dm644 dev.hdos.HDOS.png /app/share/icons/hicolor/512x512/apps/dev.hdos.HDOS.png",
```

The png is 512x512.

File: dev.hdos.HDOS.json [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                    "type": "script",
+                    "dest-filename": "hdos",
+                    "commands": [
+                        "for file in $XDG_RUNTIME_DIR/app/com.discordapp.Discord/discord-ipc-*; do ln -sf $file $XDG_RUNTIME_DIR/$(basename $file); done",
+                        "Exec=env vblank_mode=0 $JAVA_HOME/bin/java -jar /app/share/hdos-launcher.jar \"$@\""
+                    ]
+                },
+                {
+                    "type": "file",
+                    "url": "https://cdn.hdos.dev/launcher/latest/hdos-launcher.jar",
```
Comment (hfiguiere) [inline dev.hdos.HDOS.json:74]: since this file isn't versioned, if upstream has a new version, this will break.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by Brainfree** [comment]
> Submission PR should be against the `new-pr` branch.

I changed the base branch, applying suggested fixes. I'll submit a correct one.

File: dev.hdos.HDOS.appdata.xml (hunk 1 above)
Comment (Brainfree) [reply dev.hdos.HDOS.appdata.xml:3]: Done

File: dev.hdos.HDOS.json (hunk 2 above)
Comment (Brainfree) [reply dev.hdos.HDOS.json:20]: Yeah, thought that may be a problem, I'll tweak the it

File: dev.hdos.HDOS.json (hunk 3 above)
Comment (Brainfree) [reply dev.hdos.HDOS.json:47]: Deleted

File: dev.hdos.HDOS.json (hunk 4 above)
Comment (Brainfree) [reply dev.hdos.HDOS.json:59]: Fixed, nice catch

Comment (Brainfree) [reply dev.hdos.HDOS.json:59]: I'll use the smaller one instead

File: dev.hdos.HDOS.json (hunk 5 above)
Comment (Brainfree) [reply dev.hdos.HDOS.json:74]: I'll version it

**Comment by Brainfree** [comment]
Applied fixes and opened it against the new-pr [here](https://github.com/flathub/flathub/pull/3628)

**Pull request closed without merging**